### PR TITLE
feat(dsg): add util methods to ast.ts

### PR DESCRIPTION
**Pull request by yuval-hazaz**
Issue Number: #2556 

## PR Details

This PR adds two new util methods to help to manipulate decorators in the DSG

## PR Checklist
- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/code_of_conduct.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by yuval-hazaz** [comment]
@tupe12334 @mshidlov 
Please see if you can suggest any improvement to the implementation of the new functions 
They will be used in another PR, but I included them in a separate PR so we can review it from a generic perspective first

File: packages/amplication-data-service-generator/src/util/ast.ts [hunk 1]
```diff
@@ -769,3 +769,70 @@ export function createGenericArray(
...
+      return this.traverse(path);
+    },
+    // Recast has a bug of traversing class property decorators
+    // This method fixes it
+    visitClassProperty(path) {
+      const childPath = path.get("decorators");
+      if (childPath.value) {
+        this.traverse(childPath);
+      }
+      this.traverse(path);
```
Comment (abrl91) [inline packages/amplication-data-service-generator/src/util/ast.ts:807]: don't need to return here?

Comment (yuval-hazaz) [reply packages/amplication-data-service-generator/src/util/ast.ts:807]: The reason we implement visitClassProperty and visitClassDeclaration is to make the visit method traverse into decorators - there is a bug in the package.
So the idea is just to check if there are any decorators to the class or property and traverse into it

**Comment by tupe12334** [comment]
@yuval-hazaz is it not better to export those functions to separated files for more readability?

**Comment by yuval-hazaz** [comment]
> @yuval-hazaz is it not better to export those functions to separated files for more readability?

Thanks Ofek, it may be better - we can consider extracting all functions to separate files in a dedicated folder. Actually, in the following PR #2558 you can see that one of the function was already moved to another PR

**Comment by tupe12334** [comment]
@yuval-hazaz Please check if you need the branch and if not please delete it to prevent piling up branches

**Comment by yuval-hazaz** [comment]
thansk @tupe12334, done

**Pull request closed without merging**